internal/api/token.go

```diff
@@ -361,15 +357,10 @@ func (a *API) updateMFASessionAndClaims(r *http.Request, tx *storage.Connection,
 		return nil, internalServerError("Cannot read SessionId claim as UUID").WithInternalError(err)
 	}
 	err = tx.Transaction(func(tx *storage.Connection) error {
-		session, terr := models.FindSessionByID(tx, sessionId, false)
-		if terr != nil {
+		if terr := models.AddClaimToSession(tx, sessionId, authenticationMethod); terr != nil {
 			return terr
 		}
-		terr = models.AddClaimToSession(tx, session, authenticationMethod)
-		if terr != nil {
-			return terr
-		}
-		session, terr = models.FindSessionByID(tx, sessionId, false)
+		session, terr := models.FindSessionByID(tx, sessionId, false)
```

Comment:
Actually I did see this a few days ago, but I think this is intentional. First the claims are added to the session row on 368 and the query on 363 is to load the changes from the database. So I don't think this should be merged without making sure that 368 modifies the session object here correctly.

Reply:
as discussed internally, this isn't an issue because they are in the same transaction